workflow-manager/limiter/limiter_test.go

```diff
@@ -0,0 +1,42 @@
+package limiter
+
+import (
+	"testing"
+	"time"
+)
+
+func TestTicketDoubleDone(t *testing.T) {
+	l := New(5)
+	l.Execute(func(ticket *Ticket) {
+		l.Done(ticket)
+		l.Done(ticket)
+	})
+	t.Log("l.Execute did not block")
+}
+
+func TestWorkerCount(t *testing.T) {
+	l := New(5)
+	counterChannel := make(chan int)
+	for i := 0; i < 10; i++ {
+		go l.Execute(func(ticket *Ticket) {
+			counterChannel <- 1
+		})
+	}
+	counter := 0
+
+	// We're relying on go finishing 5 goroutines within a second
+	// This shouldn't be something we do in actual code but for test cases...eh it's fine?
+causality:
+	for {
+		select {
+		case <-counterChannel:
+			counter++
+		case <-time.After(time.Second):
+			break causality
+		}
+	}
```

Comment:
I worry that this test is fragile due to the timing issue you mention, forces a one second delay in the test suite (these things add up as test suites grow) and doesn't prove that the remaining five jobs ever get to execute once the first five complete.

[Here's a Go playground](https://play.golang.org/p/UnMlkPcyg0U) that I think implements a more thorough test in its `main()` function. Note that we use signaling between the function and job goroutines to guarantee that we're blocked on the ticket from the first job being returned, and then verify that the other two jobs get to run after the first completes. There is still a 10 millisecond wait (`case <-time.After(10 * time.Millisecond)` on line 152) that we use to introduce a window of time during which we don't expect the second and third jobs to run, but I'm not sure how to get rid of that without doing some kind of Goroutine inspection that I don't know how to do.

I left in a ton of comments and `log` statements to make it obvious what is happening; you may wish to delete those if you opt to incorporate my test.